cheroot/server.py

```diff
@@ -1791,19 +1790,15 @@ def prepare(self):  # noqa: C901  # FIXME
 
     def serve(self):
         """Serve requests, after invoking :func:`prepare()`."""
-        self.serving = True
-        while self.ready:
-            try:
-                self.tick()
-            except (KeyboardInterrupt, SystemExit):
-                raise
-            except Exception:
-                self.error_log(
-                    'Error in HTTPServer.tick', level=logging.ERROR,
-                    traceback=True,
-                )
-
-        self.serving = False
+        try:
+            self.connections.run()
+        except (KeyboardInterrupt, SystemExit):
+            raise
+        except Exception:
+            self.error_log(
+                'Error in HTTPServer.serve', level=logging.ERROR,
+                traceback=True,
+            )
```

Comment:
Is this a breaking change? It probably requires updating the module docstring that contains some lifecycle pointers.